diff --git a/cores/arduino/ch32/backup.h b/cores/arduino/ch32/backup.h
index 8eb6c3f..f25277d 100644
--- a/cores/arduino/ch32/backup.h
+++ b/cores/arduino/ch32/backup.h
@@ -40,8 +40,10 @@ static inline void resetBackupDomain(void)
 #ifdef PWR_MODULE_ENABLED
   PWR_BackupAccessCmd(ENABLE);
 #endif
+#ifndef CH32V00x
   RCC_BackupResetCmd(ENABLE);
   RCC_BackupResetCmd(DISABLE);
+#endif
 }
 
 static inline void enableBackupDomain(void)
diff --git a/cores/arduino/ch32/lib/ch32yyxx_can.c b/cores/arduino/ch32/lib/ch32yyxx_can.c
index 320ac14..4b69d94 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_can.c
+++ b/cores/arduino/ch32/lib/ch32yyxx_can.c
@@ -2,6 +2,6 @@
 #include "ch32v20x_can.c"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_can.c"
 #endif
\ No newline at end of file
diff --git a/cores/arduino/ch32/lib/ch32yyxx_can.h b/cores/arduino/ch32/lib/ch32yyxx_can.h
index 271de97..4fa5143 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_can.h
+++ b/cores/arduino/ch32/lib/ch32yyxx_can.h
@@ -5,7 +5,7 @@
 #include "ch32v20x_can.h"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_can.h"
 #endif
 
diff --git a/cores/arduino/ch32/lib/ch32yyxx_crc.c b/cores/arduino/ch32/lib/ch32yyxx_crc.c
index eb32fbb..2fe2c6a 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_crc.c
+++ b/cores/arduino/ch32/lib/ch32yyxx_crc.c
@@ -6,6 +6,6 @@
 #include "ch32v20x_crc.c"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_crc.c"
 #endif
\ No newline at end of file
diff --git a/cores/arduino/ch32/lib/ch32yyxx_crc.h b/cores/arduino/ch32/lib/ch32yyxx_crc.h
index a4a5a0b..70041d1 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_crc.h
+++ b/cores/arduino/ch32/lib/ch32yyxx_crc.h
@@ -9,7 +9,7 @@
 #include "ch32v20x_crc.h"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_crc.h"
 #endif
 
diff --git a/cores/arduino/ch32/lib/ch32yyxx_it.h b/cores/arduino/ch32/lib/ch32yyxx_it.h
index 1bfb31c..169c3f7 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_it.h
+++ b/cores/arduino/ch32/lib/ch32yyxx_it.h
@@ -17,7 +17,7 @@
 #include "ch32v20x_it.h"
 #endif
 
-#ifdef defined (CH32V30x) || defined(CH32V30x_C)
+#if defined (CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_it.h"
 #endif
 
diff --git a/cores/arduino/ch32/lib/ch32yyxx_opa.c b/cores/arduino/ch32/lib/ch32yyxx_opa.c
index 3dff6be..65a1c28 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_opa.c
+++ b/cores/arduino/ch32/lib/ch32yyxx_opa.c
@@ -11,6 +11,6 @@
 #include "ch32v20x_opa.c"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_opa.c"
 #endif
diff --git a/cores/arduino/ch32/lib/ch32yyxx_opa.h b/cores/arduino/ch32/lib/ch32yyxx_opa.h
index ae29801..93c1b24 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_opa.h
+++ b/cores/arduino/ch32/lib/ch32yyxx_opa.h
@@ -14,7 +14,7 @@
 #include "ch32v20x_opa.h"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_opa.h"
 #endif
 
diff --git a/cores/arduino/ch32/lib/ch32yyxx_pwr.c b/cores/arduino/ch32/lib/ch32yyxx_pwr.c
index 8afbfa0..c897dfe 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_pwr.c
+++ b/cores/arduino/ch32/lib/ch32yyxx_pwr.c
@@ -14,6 +14,6 @@
 #include "ch32v20x_pwr.c"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_pwr.c"
 #endif
\ No newline at end of file
diff --git a/cores/arduino/ch32/lib/ch32yyxx_rcc.h b/cores/arduino/ch32/lib/ch32yyxx_rcc.h
index c8c4099..ddab956 100644
--- a/cores/arduino/ch32/lib/ch32yyxx_rcc.h
+++ b/cores/arduino/ch32/lib/ch32yyxx_rcc.h
@@ -17,7 +17,7 @@
 #include "ch32v20x_rcc.h"
 #endif
 
-#ifdef defined(CH32V30x) || defined(CH32V30x_C)
+#if defined(CH32V30x) || defined(CH32V30x_C)
 #include "ch32v30x_rcc.h"
 #endif
 
diff --git a/cores/arduino/ch32/uart.c b/cores/arduino/ch32/uart.c
index eaf94c6..5116b0a 100644
--- a/cores/arduino/ch32/uart.c
+++ b/cores/arduino/ch32/uart.c
@@ -89,6 +89,8 @@ static serial_t serial_debug =
   .index = UART_NUM,
 };
 
+extern uint64_t GetTick(void);
+
 /* Aim of the function is to get serial_s pointer using huart pointer */
 /* Highly inspired from magical linux kernel's "container_of" */
 serial_t *get_serial_obj(UART_HandleTypeDef *huart)
@@ -263,7 +265,7 @@ void uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t par
 
   if (uart_rx == NP) 
   {
-      USART_HalfDuplexCmd(huart,ENABLE);
+      USART_HalfDuplexCmd(huart->Instance,ENABLE);
   }
 
   USART_Init(huart->Instance, &(huart->init));
